core/src/main/scala/kafka/server/QuotaFactory.scala

```diff
@@ -0,0 +1,62 @@
+package kafka.server
+
+import kafka.common.TopicAndPartition
+import org.apache.kafka.common.metrics.Metrics
+import org.apache.kafka.common.utils.SystemTime
+
+import scala.collection.Map
+
+object QuotaType  {
+  case object Fetch extends QuotaType
+  case object Produce extends QuotaType
+  case object LeaderReplication extends QuotaType
+  case object  FollowerReplication extends QuotaType
+}
+sealed trait QuotaType
+
+object QuotaFactory {
+
+
+  object UnbreakableQuota extends ReadOnlyQuota {
+    override def bound(): Int = Int.MaxValue
+
+    override def isThrottled(topicAndPartition: TopicAndPartition): Boolean = false
+
+    override def isQuotaExceededBy(bytes: Int): Boolean = false
+  }
+
+  case class QuotaManagers(client: Map[QuotaType, ClientQuotaManager], leaderReplication: ReplicationQuotaManager, followerReplication: ReplicationQuotaManager)
```

Comment:
This class seems a bit inconsistent in that it groups the client replication types in a map, but not the replication ones. It seems like it might be better to either have them all as fields or all in maps. Is there a reason to do it this way instead?